### PR TITLE
Harmony 1741 - Support for sending variables via query params or post body

### PR DESCRIPTION
**Pull request by indiejames**
## Jira Issue ID
HARMONY-1741

## Description
This PR adds support for specifying variables as either query parameters or as part of the post body using a comma separated list.

## Local Test Steps

1. After deploying locally (or in sandbox) use the following query (note the use of the 'parameter_vars' value where the variable(s) would normally be, and the actual vars listed in query param 'variable'.

http://localhost:3000/C1233800302-EEDTEST/ogc-api-coverages/1.0.0/collections/parameter_vars/coverage/rangeset/?subset=lat(20%3A60)&subset=lon(-140%3A-50)&outputCrs=EPSG%3A31975&format=image%2Fpng&maxResults=1&variable=blue_var,red_var,green_var

2. (optional) run the same query using a POST and specifying the query params in the body
3. Verify the online docs are updated to mention 'parameter_vars'

## PR Acceptance Checklist
* [x] Acceptance criteria met
* [x] Tests added/updated (if needed) and passing
* [x] Documentation updated (if needed)

### CODE REVIEW

File: services/harmony/test/ogc-api-coverages/get-coverage-rangeset.ts [hunk 1]
```diff
@@ -343,6 +343,127 @@ describe('OGC API Coverages - getCoverageRangeset', function () {
     });
   });
 
+  describe('Using the "parameter_vars" pseudo-variable', function () {
```
Comment (ygliuvt) [inline services/harmony/test/ogc-api-coverages/get-coverage-rangeset.ts:346]: We should add a test for `"parameter_vars" specified, but no variables given`

**Review by ygliuvt — APPROVED** [review]
Tested successfully locally. Minor comments on adding a test case.

File: services/harmony/app/util/variables.ts [hunk 2]
```diff
@@ -129,10 +131,26 @@ export function parseVariables(
...
+        collectionId: collection.id, shortName: collection.short_name,
+        versionId: collection.version_id, coordinateVariables,
+      });
     }
   } else {
+    if (variableIds.indexOf('parameter_vars') !== -1) {
+      if (!queryVars) {
+        throw new RequestValidationError('"parameter_vars" specified, but no variables given');
+      } else {
+        // variableIds = [(queryVars as string | string[])].flat();
```
[the remark this answers is not in the document]
Comment (indiejames) [reply services/harmony/app/util/variables.ts:144]: Done.

File: services/harmony/test/ogc-api-coverages/get-coverage-rangeset.ts (hunk 1 above)
Comment (indiejames) [reply services/harmony/test/ogc-api-coverages/get-coverage-rangeset.ts:346]: Good idea. Done.

File: services/harmony/test/ogc-api-coverages/get-coverage-rangeset-shapefile.ts [hunk 3]
```diff
@@ -54,6 +54,42 @@ describe('OGC API Coverages - getCoverageRangeset with shapefile', function () {
...
+    };
+
+    describe('and a valid shapefile', function () {
+      const shapeForm = { ...form, shapefile: { path: './test/resources/complex_multipoly.geojson', mimetype: 'application/geo+json' } };
+      StubService.hook({ params: { redirect: 'http://example.com' } });
+      cmrResp.headers = new fetch.Headers(cmrResp.headers);
+      hookPostRangesetRequest(version, collection, 'parameter_vars', shapeForm);
+
+      it('passes the source collection to the backend', function () {
+        console.log(`${JSON.stringify(this.res, null, 2)}`);
```
Comment (chris-durbin) [inline services/harmony/test/ogc-api-coverages/get-coverage-rangeset-shapefile.ts:76]: Remove the `console.log`

**Review by chris-durbin — APPROVED** [review]
Tested successfully with a POST.

File: services/harmony/test/ogc-api-coverages/get-coverage-rangeset-shapefile.ts (hunk 3 above)
Comment (indiejames) [reply services/harmony/test/ogc-api-coverages/get-coverage-rangeset-shapefile.ts]: Done.

**Pull request merged**